### PR TITLE
kafkasource: change bootstrapServers and Topics to accept arrays

### LINKED ISSUE #1055: Kafka source - comma or array for servers and topics
**Problem**

Today, for the Kafka event source, when people specify the list of bootstrap servers or topics, they're comma separated lists. When these lists get long (e.g. I'm messing with a usecase with ~100 topics), I'm wondering if a real array wouldn't be easier for people to manage.  And for consistency, I have the same question for the list of bootstrap servers.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**

Event consumer

### PR DESCRIPTION
**Pull request by lberk**
Each item in the list can be comma separated

- This fixes #1055

**Release Note**
```Action Required: KafkaSource now requires bootstrapServers and Topics to be in lists```

### CODE REVIEW

**Comment by lionelvillard** [comment]
/hold

This change is not backward compatible.

**Comment by lberk** [comment]
@lionelvillard right, I'm open to suggestions on how to do so -- but keeping in mind that the kafka api is `v1alpha1` we're not actually beholden (yet) to keeping it a plain string.

Keep in mind that this change allows users to keep comma separated values in each element in the list, so it would be a minimal change.

**Comment by aslom** [comment]
This will be backward incompatible change for kafka source in knative eventing.

Not sure about how YAML parsing is done - is it possible to support both string and sequence of strings?

**Comment by matzew** [comment]
I'd also not block this, due to the fact that kafka is _early_ :man_shrugging:

**Comment by lionelvillard** [comment]
/unhold

I'm fine with just having an `Action Required` in the release section.

Can you also open a doc issue to update the relevant Kafka examples?

File: kafka/source/README.md [hunk 1]
```diff
@@ -38,8 +38,10 @@ event sink.
      consumerGroup: knative-group
      # Broker URL. Replace this with the URLs for your kafka cluster,
      # which is in the format of my-cluster-kafka-bootstrap.my-kafka-namespace:9092.
-     bootstrapServers: REPLACE_WITH_CLUSTER_URL
-     topics: knative-demo-topic
+     bootstrapServers:
+     - REPLACE_WITH_CLUSTER_URL
```
Comment (mattmoor) [inline kafka/source/README.md:42]: Format markdown:
```suggestion
       - REPLACE_WITH_CLUSTER_URL
```

File: kafka/source/README.md [hunk 2]
```diff
@@ -38,8 +38,10 @@ event sink.
...
+     topics:
+     - knative-demo-topic
```
Comment (mattmoor) [inline kafka/source/README.md:44]: Format markdown:
```suggestion
       - knative-demo-topic
```

**Comment by lberk** [comment]
@lionelvillard done, I've opened knative/docs#2410 to update the docs instead of filing an issue.  Also updated the README.md in kafka's own source
thanks!

**Pull request merged**